packages/front-end/hooks/useStripeSubscription.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from "react";
+import { OrganizationInterface } from "../../back-end/types/organization";
+import { useAuth } from "../services/auth";
+import useApi from "./useApi";
+
+export default function useStripeSubscription() {
+  const { apiCall } = useAuth();
+  const { data } = useApi<{
+    organization: OrganizationInterface;
+  }>(`/organization`);
+  const [upcomingInvoice, setUpcomingInvoice] = useState(null);
+  const [pricePerSeat, setPricePerSeat] = useState(null);
+
+  useEffect(() => {
+    const getPriceData = async () => {
+      const { upcomingInvoice } = await apiCall(`/upcoming-invoice`);
+      if (upcomingInvoice) {
+        setUpcomingInvoice(upcomingInvoice);
+      }
+      const { priceData } = await apiCall(`/subscription-data`);
```

Comment:
@jdorn Is there a better way to do this? Currently, if someone has a valid subscription and is adding an additional user, the price takes a while to render within the `InviteModalSubscriptionInfo` component which is a bit weird. I tried to memoize this, but useMemo is returning a promise.